### PR TITLE
🎉Source MailChimp: New Stream Automations

### LINKED ISSUE #17024: Source Mailchimp: new stream Automations
## Tell us about the problem you're trying to solve
[Automations](https://mailchimp.com/developer/marketing/api/automation/)
Mailchimp's classic automations feature lets you build a series of emails that send to subscribers when triggered by a specific date, activity, or event. Use the API to manage Automation workflows, emails, and queues. Does not include Customer Journeys.

## Describe the solution you’d like
Research pros and cons for implementing this stream.
_Research should contain: what stream should be implemented, permissions and scopes (if needed), paid plan (if needed)._

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
Closes https://github.com/airbytehq/airbyte/issues/17024

## Enhancement
Adds a new stream: 

- Automations (Incremental)

## Recommended reading order
1. `source_mailchimp/source.py`
2. `source_mailchimp/streams.py`
3.  `source_mailchimp/schemas/automations.json`

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Checklist
<details open><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter [Yes]

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [x] Documentation that references the generator is updated as needed

</details>

## For the reviewer: 
- New stream is working in my local system and all acceptance test passing
- Please review & test it on the prod

### CODE REVIEW

**Comment by btkcodedev** [comment]
@YowanR could you please assign anyone for review

**Comment by natalyjazzviolin** [comment]
@btkcodedev could you please write out what issue this is solving in the future instead of just linking the issue? That would be really helpful, thank you :)

**Comment by natalyjazzviolin** [comment]
@btkcodedev Please make sure to run `./gradlew --no-daemon  :airbyte-integrations:connectors:source-mailchimp:airbytePythonFormat` to fix formatting errors.

**Comment by btkcodedev** [comment]
@natalyjazzviolin Thanks for the comments and build, 
Primary keys, and cursor fields, all are updated and schema fixed.
Acceptance test is passing in local
Please continue the further review.

**Comment by btkcodedev** [comment]
@natalyjazzviolin Bump here

**Comment by btkcodedev** [comment]
Ready for review :)

**Comment by btkcodedev** [comment]
@natalyjazzviolin Bump Here :)

**Comment by btkcodedev** [comment]
@natalyjazzviolin @sh4sh @marcosmarxm @YowanR 
The PR is ready for review, 
Please could anyone run the test command :)

**Comment by btkcodedev** [comment]
The Docker version is set to 5.0 as the older version 3.5 was a stream update so considering it as a minor update rather than a patch,
Old -> 4.0 and now it's 5.0

**Comment by marcosmarxm** [comment]
Sorry @btkcodedev for the delay to review the contribution. @airbytehq/community-assistance-dri will review during this week.

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/spec.json [hunk 1]
```diff
@@ -38,6 +38,12 @@
                 "type": "string",
                 "description": "An access token generated using the above client ID and secret.",
                 "airbyte_secret": true
+              },
+              "server_prefix": {
+                "title": "server_prefix",
+                "type": "string",
+                "description": "Server prefix is attached to beginning of every endpoint ping.",
+                "airbyte_secret": true
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailchimp/source_mailchimp/spec.json:46]: Why are you adding this? The PR is adding a new stream not changing this part of the code. Please rollback the change

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/streams.py [hunk 2]
```diff
@@ -129,6 +129,14 @@ def path(self, **kwargs) -> str:
         return "campaigns"
 
 
+class Automations(IncrementalMailChimpStream):
+    cursor_field = "create_time"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailchimp/source_mailchimp/streams.py:133]: Add doc link to Mailchimp docs api.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/streams.py (hunk 2 above)
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-mailchimp/source_mailchimp/streams.py:134]: Added

**Comment by sajarin** [comment]
Hey thanks for the contribution and apologies for the delay! 

I'll be the DRI responsible for reviewing this and getting it across the finish line. We currently have a backlog of about a dozen PRs, but just wanted to let you know that this PR has been added to the queue and I'll be trying my best to get to this as soon as possible. 

Thanks for the contribution and for being patient :)

**Comment by btkcodedev** [comment]
Thanks, It is ready for review,
Please forward

**Review by marcosmarxm — APPROVED** [review]
Thanks @btkcodedev I requested to the connector team to made the final review of the contribution. Hope to publish during the week.

**Comment by btkcodedev** [comment]
Thanks @marcosmarxm for update

**Review by arsenlosenko — APPROVED** [review]

**Pull request merged**